subsys/debug/tracing/tracing_backend_uart.c

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2019 Intel corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <ctype.h>
+#include <kernel.h>
+#include <device.h>
+#include <assert.h>
+#include <drivers/uart.h>
+#include <tracing_core.h>
+#include <tracing_buffer.h>
+#include <tracing_backend.h>
+
+static struct device *dev;
+
+#ifdef CONFIG_TRACING_HANDLE_HOST_CMD
+static void uart_isr(struct device *dev)
+{
+	int rx;
+	u8_t byte;
+	static u8_t *cmd;
+	static u32_t length, cur;
+
+
+	while (uart_irq_update(dev) && uart_irq_is_pending(dev)) {
+		if (!uart_irq_rx_ready(dev)) {
+			continue;
+		}
+
+		rx = uart_fifo_read(dev, &byte, 1);
+		if (rx < 0) {
+			uart_irq_rx_disable(dev);
+			return;
+		}
+
+		if (!cmd) {
+			length = tracing_cmd_buffer_alloc(&cmd);
+			if (!cmd) {
+				return;
+			}
+		}
+
+		if (!isprint(byte)) {
+			switch (byte) {
```

Comment:
this switch could be an `if` since there is checking for only one possible case. (MISRA-C has rules related with it)

Reply:
thanks, updated